src/desert/__init__.py

```diff
@@ -63,7 +67,10 @@ class A:
     """
     meta = metadata(marshmallow_field)
     meta.update(kw.pop("metadata", {}))
-    return dataclasses.field(**kw, metadata=meta)
+    # typeshed hints it as Mapping[str, Any] without any obvious reason
+    # https://github.com/python/typeshed/blob/95a45eb4abd0c25849268983cb614e3bf6b9b264/stdlib/dataclasses.pyi#L81
+    # https://github.com/python/typeshed/pull/5823
+    return dataclasses.field(**kw, metadata=meta)  # type: ignore[arg-type]
```

Comment:
Until it gets merged and mypy gets released.  Then depending on mypy settings we might get a failure reminding us to take it out.  Otherwise we have to not check hints or have failing CI.